internal/core/server/kernel.go

```diff
@@ -200,3 +200,32 @@ func (kern *Kernel) DescribeTasks(ctx context.Context, input *api.DescribeTasksI
 	}
 	return &output, nil
 }
+
+func (kern *Kernel) GetUserHomeDir(ctx context.Context, input *api.GetUserHomeDirInput) (*api.GetUserHomeDirOutput, error) {
+	path, err := os.UserHomeDir()
+	if err != nil {
+		return &api.GetUserHomeDirOutput{}, errors.New("could not get user home directory")
+	}
+	return &api.GetUserHomeDirOutput{Path: path}, nil
+}
+
+func (kern *Kernel) ReadDir(ctx context.Context, input *api.ReadDirInput) (*api.ReadDirOutput, error) {
+	if !path.IsAbs(input.Path) {
```

Comment:
As best I can tell, every usage of `path` in this file should actually be `filepath`. The `path` package is for URLs, idealized paths, etc. The `filepath` package will use the correct separator for the current OS. The two packages have largely identical interfaces otherwise.

Reply:
:+1: ah cheers!